src/pages/settings/index.tsx

```diff
@@ -33,8 +32,8 @@ export default function SettingsPage(): ReactElement {
 
       // We can't set the RPC URL to the `swap-endpoint` Bee config value unless the Bee node is already in
       // light mode as setting this config value, basically upgrades the node to light mode.
-      if (isDesktop && nodeInfo?.beeMode === BeeModes.LIGHT) {
```

Comment:
The underlying issue is that Bee may not start due to having a bad RPC endpoint. Then we cannot rely on the `beeMode` check.

I'll use the desktop config API instead..

Reply:
I see, right, then we need to make sure we won't accidentally upgrade the Bee with just setting RPC... It is a bit of a mess since we use RPC also in Dashboard, but then it should not be set in Bee until the user wants to upgrade Bee 🙄